go/examples/webhook-server/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	tlsigning "github.com/Truelayer/truelayer-signing/go"
+	"github.com/gregjones/httpcache"
+)
+
+func main() {
+	tp := httpcache.NewMemoryCacheTransport()
+	client := http.Client{Transport: tp}
+
+	http.HandleFunc("/hook/d7a2c49d-110a-4ed2-a07d-8fdb3ea6424b", receiveHook(&client))
+
+	log.Println("Starting server on: 7000")
+
+	log.Fatal(http.ListenAndServe(":7000", nil))
+}
+
+func receiveHook(client *http.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			http.Error(w, "Method is not supported.", http.StatusNotFound)
+			return
+		}
+
+		err := verifyHook(client, r)
+		if err != nil {
+			log.Println(err)
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+
+		// handle verified hook
+
+		w.WriteHeader(http.StatusAccepted)
+	}
+}
+
+func verifyHook(client *http.Client, r *http.Request) error {
+	tlSignature := r.Header.Get("Tl-Signature")
+	if len(tlSignature) == 0 {
+		return fmt.Errorf("missing Tl-Signature header")
+	}
+
+	jwsHeader, err := tlsigning.ExtractJwsHeader(tlSignature)
+	if err != nil {
+		return fmt.Errorf("jku missing")
```

Comment:
Why don't we return the `err` itself in here? It already has useful error message.

Reply:
Good point!